spec/services/reports/merits_report_creator_spec.rb

```diff
@@ -30,5 +31,53 @@
       create :attachment, :merits_report, legal_aid_application: legal_aid_application
       expect { subject }.not_to change { Attachment.count }
     end
+
+    context 'ccms case ref does not exist' do
+      let(:legal_aid_application) do
+        create :legal_aid_application,
+               :with_application_proceeding_type,
+               :with_lead_proceeding_type,
+               :with_everything,
+               :generating_reports,
+               ccms_submission: ccms_submission
+      end
+      let(:ccms_submission) { create :ccms_submission }
+
+      before do
+        allow_any_instance_of(CCMS::Submission).to receive(:process!).with(any_args).and_return(true)
+      end
+
+      it 'processes the existing ccms submission' do
+        expect(legal_aid_application.reload.ccms_submission).to receive(:process!)
+        subject
+      end
+    end
+
+    context 'ccms submission does not exist' do
+      let(:legal_aid_application) do
+        create :legal_aid_application,
+               :with_application_proceeding_type,
+               :with_lead_proceeding_type,
+               :with_everything,
+               :generating_reports
+      end
+      let(:ccms_submission) { create :ccms_submission }
+
+      before do
+        RSpec::Mocks.configuration.allow_message_expectations_on_nil = true
```

Comment:
Please could you have a look at this

I added .allow_message_expectations_on_nil  because create_ccms_submission wasn't creating the ccms submission because it's mocked out and the submission was nil thus being `nil.process!` . I thought if it at least tests that the process! is being called then that's ok.

Reply:
Let's have a chat about this one...